distributed/shuffle/_shuffle_extension.py

```diff
@@ -588,29 +684,111 @@ def get(
             self.columns[id] = column
             self.output_workers[id] = output_workers
             self.completed_workers[id] = set()
+            self.participating_workers[id] = output_workers.copy()
```

Comment:
why copy? This is a newly created set in this method

Reply:
We assign `output_workers` to `self.output_workers[id]` two lines and we need these to be two independent sets.